PictureApp/Controller/PicturesViewController.swift

```diff
@@ -0,0 +1,306 @@
+        selectedCategoryForSort = selectedCategory
+    }
+    
+    
+    //MARK:Delegate for order
+    func optionsViewController(_ controller: OptionsViewController, didSelectOrderStrategy order: Order) {
+        currentPage = 1
+        hits.removeAll()
+        fetchHits(order: order , selectedCategory: selectedCategoryForSort,currentPage: currentPage)
+          DispatchQueue.main.async{
+            self.collectionView.reloadData()
+        }
+        navigationController?.popViewController(animated: true)
+        orderForCategories = order
+    }
+   
+    
+    func fetchHits(order :Order , selectedCategory :[String], currentPage :Int) {
+       
+        PictureService(with: URLHTTPClient(session: URLSession.shared))
+            .fetch(completion: {post , totalHits in
+                self.hits.append(contentsOf:post)
+                self.totalHits = totalHits
+            }, order: order, selectedCategory: selectedCategory,currentPage: currentPage)
+      
+       
+    }
+    
+    
+    //MARK:Custom dropDownButton
+    @IBAction func showOptions(_ sender: UIButton) {
+        
+        let  optionsVc = self.storyboard?.instantiateViewController(withIdentifier: "OptionsViewController") as! OptionsViewController
+        optionsVc.modalPresentationStyle = .popover
+        let popoverVc = optionsVc.popoverPresentationController!
+        popoverVc.permittedArrowDirections = .up
+        popoverVc.sourceRect = sender.convert(sender.frame, to: nil)
+        optionsVc.delegate = self
+        popoverVc.delegate = self
+        popoverVc.sourceView = view
+            
+        
+        
+        present(optionsVc, animated: true, completion: nil)
+    
+    }
+    
+    
+    //MARK:Delegate for filter
+    func didFilterSelected(){
+        let filterVc = self.storyboard?.instantiateViewController(withIdentifier: "FilterViewController")  as! FilterViewController
+        filterVc.delegate = self
+        filterVc.title = "Select a Category"
+        
+        navigationController?.pushViewController(filterVc, animated: true)
+        
+        
+    }
+    
+    
+    func setupCollectionView(){
+        
+        collectionView.register(GalleryCollectionViewCell.self, forCellWithReuseIdentifier: GalleryCollectionViewCell.identifier)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        setupCollectionViewLayout()
+        view.addSubview(collectionView)
+        let margins = view.layoutMarginsGuide
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.leadingAnchor.constraint(equalTo: margins.leadingAnchor).isActive = true
+        collectionView.bottomAnchor.constraint(equalTo: margins.bottomAnchor).isActive = true
+        collectionView.trailingAnchor.constraint(equalTo: margins.trailingAnchor).isActive = true
+        collectionView.topAnchor.constraint(equalTo: margins.topAnchor).isActive = true
+        
```

Comment:
Don't use a new lines at the end of the method

Reply:
Done